### PR TITLE
adding error messages to form

### PR DESCRIPTION
**Pull request by Ibabalola**

### CODE REVIEW

File: app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx [hunk 1]
```diff
@@ -14,7 +18,46 @@ export const loader: LoaderFunction = async ({ params, request }: DataFunctionAr
...
+    "speciesCode": species,
+    "state": state,
+    "stateLabel":"",
+    "scientificName":"",
+    "commodity_code": commodityCode,
+    "commodity_code_description":""
+  }
+  
+  const response = await addSpecies(catchCertificate, resquestBody);
+  const errors: IError[] = response.errors || [];
```
Comment (daniele-zurico) [inline app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx:46]: I think I start seeing this in more than one place... shall we put in a common place?

File: app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting/whatAreYouExporting.ts [hunk 2]
```diff
@@ -145,4 +148,45 @@ export const getAddSpeciesLoaderData = async (catchCertificate: string, faoCode:
...
     presentationCode
   }
+}
+
+export const addSpecies = async (catchCertificate: string | undefined, requestBody: any): Promise<any> => {
+  if(!catchCertificate) {
+    throw new Error("catchCertificate is required");
+  }
+
+  const response = await fetch(ADD_SPECIES_URL,
```
Comment (daniele-zurico) [inline app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting/whatAreYouExporting.ts]: this post is in more then one place and is quite verbose... does make sense to create a generic wrapper in common and just call it where we need?

Comment (Ibabalola) [reply app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting/whatAreYouExporting.ts]: Updated @daniele-zurico to use a reusable function called `apiCall: (url: string, requestHeaders?: HeadersInit,  method?: string, requestBody?: any) => Promise<Response>`

File: app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx (hunk 1 above)
Comment (Ibabalola) [reply app/routes/create-catch-certificate/$catchCertificate/what-are-you-exporting.tsx:46]: Updated @daniele-zurico, we now have this function `export const apiCallFailed: (errors: IError[], data?: any) => Response` to centralise this behaviour.

File: app/communication/index.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { isEmpty } from "lodash";
+import { json } from "remix";
+import { IError } from "~/types";
+import { getTransformedError } from "~/helpers";
+
+export const apiCall: (url: string, requestHeaders?: HeadersInit,  method?: string, requestBody?: any) => Promise<Response> = async (url: string, requestHeaders?: HeadersInit, method: string = "GET", requestBody?: any): Promise<Response> =>
```
Comment (daniele-zurico) [inline app/communication/index.ts]: it could be good if we have specific:
- post
- get
- etc
so when we use it is easy to identify what it does

**Review by daniele-zurico — APPROVED** [review]

**Pull request merged**